### DISCUSSION TITLE
[Gradle] Configuration validation doesn't occur for all parameters

### DISCUSSION THREAD

**Discussion by haydenbaker** [jreleaser/jreleaser#1639] [Q&A]
When running jreleaser tasks, it's helpful when it fails due to missing parameters (env-vars in this case). Example:
```
> JReleaser has not been properly configured.
  signing.passphrase must not be blank. Configure a value using the Gradle DSL, or define a System property jreleaser.gpg.passphrase, or define a JRELEASER_GPG_PASSPHRASE environment variable, or define a key/value pair in /Users/haydebak/.jreleaser/config.properties with a key named JRELEASER_GPG_PASSPHRASE

```

However, it doesn't validate params for maven/nexus2 - if we have it configured and don't set any of the expected vars ( JRELEASER_NEXUS2_USERNAME,  JRELEASER_NEXUS2_PASSWORD, or their many variants), it won't fail until runtime.

Is this expected?

**Comment by aalmiray** [comment]
No, there should be a failure during configuration time in this case as credentials are required.

**Reply by aalmiray** [reply]
Filed https://github.com/jreleaser/jreleaser/issues/1640
